operator/scheduler/client.go

```diff
@@ -133,15 +133,14 @@ func (s *SchedulerClient) RemoveConnection(namespace string) {
 // A smoke test allows us to quickly check if we actually have a functional grpc connection to the scheduler
 func (s *SchedulerClient) smokeTestConnection(conn *grpc.ClientConn) error {
 	grcpClient := scheduler.NewSchedulerClient(conn)
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
```

Comment:
It is unclear how this logic closes the stream? it looks to me it is just cancelling the context but the stream is still established unless `CloseSend()` is called?

Note that the stream for this test connection is not the actual stream that is being used to communicate update messages between `scheduler` and `controller`.

Reply:
As I understand, smoke test is used to check whether connection to scheduler still works. So controller should close the test stream.

`CloseSend` does not terminate the recv stream, it just claims it won't **send** any data.

To terminate a recv stream,  we can cancel the context on the client side